addons/lunch/models/lunch_supplier.py

```diff
@@ -196,6 +196,8 @@ def write(self, values):
             self.env['lunch.order'].search([('supplier_id', 'in', self.ids)]).write({'company_id': values['company_id']})
         super().write(values)
         if not CRON_DEPENDS.isdisjoint(values):
+            # flush fields to call _sql_constraints
+            self.flush()
```

Comment:
We don't need to flush everything, instead

`self.flush_model(['automatic_email_time_range'])
`

Reply:
ah, didn't pay attention to the version :innocent: